File: examples/sharing/pending-invitations.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Per-resource pending invitations — for "this specific document" or "this
// specific group" panels.
export async function pendingInvitations(
  client: JsBaoClient,
  documentId: string,
  groupType: string,
  groupId: string,
) {
  // #region example
  const docPending = await client.documents.listPendingInvitations(documentId);
  // [{ email, permission, invitationId, createdAt, expiresAt, grantedBy? }]

  const groupPending = await client.groups.listPendingInvitations(groupType, groupId);
  // [{ email, role, invitationId, createdAt, expiresAt, addedBy? }]
  // #endregion example
  return { docPending, groupPending };
}

File: examples/sharing/pending-invitations.swift
import JsBaoClient

// Per-resource pending invitations — for "this specific document" or "this
// specific group" panels.
func pendingInvitations(
  client: JsBaoClient,
  documentId: String,
  groupType: String,
  groupId: String
) async throws {
  // #region example
  let docPending = try await client.documents.listPendingInvitations(documentId: documentId)
  // [{ email, permission, invitationId, createdAt, expiresAt, grantedBy? }]

  let groupPending = try await client.groups.listPendingInvitations(groupType: groupType, groupId: groupId)
  // [{ email, role, invitationId, createdAt, expiresAt, addedBy? }]
  // #endregion example
  _ = (docPending, groupPending)
}
